### PR TITLE
initial crud-user api expose without db

### PR DESCRIPTION
**Pull request by bsse1009**
## Change Descriptions:
Create CRUD Operations for Users

## Change Requirements:
**Endpoints**-
            - GET /users - Get a list of users
            - GET /users/{id} - Get a user with particular id
            - POST /users - Create a user
            - PUT /users/{id} - Update a user with a particular id
            - DELETE /users/{id} - Delete a user with a particular id
        **Implementation Details**-
            - Username should be used as {id}
            - Username should not contain anything but english alphabets and digits
            - Username should be case insensitive (make all lowercase)
            - While getting a user , password shouldn’t be in the response (Introducing DTO concept)
            - If no user is found is should return correct status code and message
            - Always hash user’s password
        **Example User Model**:
        {
            Id : 1, 
            Username: turja
            Email: simanta.deb@cefalo.com
            Password: 123456
            CreatedAt: 12-12-12
            UpdatedAt: 12-12-12
        }

### CODE REVIEW

File: server/controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  constructor() {}
+
+  async getAllUsers(req, res) {
+    const data = await UsersRepository.findAll();
+    return res.status(200).send(data.rows);
+  }
+
+  async getUserById(req, res) {
+    const id = req.params.id;
+    const data = await UsersRepository.findById(id);
```
Comment (rafaelragib) [inline server/controllers/users.controller.js:15]: I think there should be a check for if id is undefined.

File: server/controllers/users.controller.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  async deleteUserById(req, res) {
+    const id = req.params.id;
+    const data = await UsersRepository.deleteUserById(id);
+    if(data.rowCount === 0) res.send("User Does not exist is DB!")
+    if (data) return res.status(200).send("User deleted successfully!");
+  }
+
+  async updateUserById(req, res) {
```
Comment (rafaelragib) [inline server/controllers/users.controller.js:32]: is this function complete?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "blogger-backend",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "start": "node app.js",
+    "dev": "nodemon app.js",
+    "start:prod": "NODE_ENV=production nodemon app.js"
```
Comment (rafaelragib) [inline package.json:8]: tip: we don't run nodemon on production. No need to change now. You can resolve this comment.

File: server/controllers/users.controller.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  async createUser(req, res) {
+    const {Username, Email, Password, CreatedAt, UpdatedAt} = await req.body;
+    const data = await UsersRepository.create([Username, Email, Password, CreatedAt, UpdatedAt])
+    if (data) return res.status(201).send("User created successfully!");
+  }
+
+  async deleteUserById(req, res) {
+    const id = req.params.id;
+    const data = await UsersRepository.deleteUserById(id);
```
Comment (rafaelragib) [inline server/controllers/users.controller.js:27]: same id check should be here.

File: server/ repositories/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+const UsersRepository = require('./users.repository');
+
+module.exports = {
+  UsersRepository
+};
```
Comment (edge555) [inline server/ repositories/index.js:7]: No need for this.

File: server/controllers/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+const UsersController = require('./users.controller');
+
+module.exports = {
+  UsersController
+};
```
Comment (edge555) [inline server/controllers/index.js:7]: No need for this.

File: server/routes/index.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+
+Router.use('/users', UsersRouter);
+
+/* GET home page. */
+Router.all('/', function(req, res, next) {
+  res.send("Welcome to home page!");
+});
+
+module.exports = Router;
```
Comment (edge555) [inline server/routes/index.js:15]: Put this in root directory and import everything from there.

File: server/controllers/users.controller.js (hunk 1 above)
Comment (bsse1009) [reply server/controllers/users.controller.js]: Got it, But if id  undefined, will it route in this function?

File: server/controllers/users.controller.js (hunk 2 above)
Comment (bsse1009) [reply server/controllers/users.controller.js:37]: It was not complete. Now it is complete after the last commit.

File: package.json (hunk 3 above)
Comment (bsse1009) [reply package.json:8]: Got it vaiya.

**Pull request merged**